src/js/tools/ellipse.js

```diff
@@ -43,7 +43,7 @@ define(function (require, exports, module) {
      * @constructor
      */
     var EllipseTool = function () {
-        var resetPromise = descriptor.playObject(toolLib.resetShapeTool()),
```

Comment:
This should be like the other two shape tools, and only the play object be defined outside the selectHandler.

If you call playObject outside the selectHandler, you're essentially calling this function only once in the life time of the app, and in all other instances of select handler being called, the promise will immediately return since it has been resolved.